src/library-authoring/import-course/stepper/ReviewImportDetails.tsx

```diff
@@ -199,16 +211,43 @@ export const ReviewImportDetails = ({ courseId, markAnalysisComplete }: Props) =
         return total;
       },
       0,
-    ) - finalUnssupportedBlocks;
-  }, [blockTypes, finalUnssupportedBlocks]);
+    ) - finalUnsupportedBlocks;
+  }, [blockTypes, finalUnsupportedBlocks]);
 
   /** Calculate the unsupported block percentage based on the final total blocks and unsupported blocks. */
   const unsupportedBlockPercentage = useMemo(() => {
     if (!blockTypes || !totalBlocks) {
       return 0;
     }
-    return (finalUnssupportedBlocks / (totalBlocks + finalUnssupportedBlocks)) * 100;
-  }, [blockTypes, finalUnssupportedBlocks]);
+    return (finalUnsupportedBlocks / (totalBlocks + finalUnsupportedBlocks)) * 100;
+  }, [blockTypes, finalUnsupportedBlocks]);
+
+  const limitIsExceeded = useMemo(() => (
+    libraryData?.numBlocks || 0) + (totalBlocks || 0) > (libraryBlockLimits?.maxBlocksPerContentLibrary || 0
+  ), [libraryData?.numBlocks, totalBlocks, libraryBlockLimits?.maxBlocksPerContentLibrary]);
+
+  useEffect(() => {
+    setImportIsBlocked(limitIsExceeded);
+  }, [limitIsExceeded, setImportIsBlocked]);
+
+  // If the total blocks exceeds the permitted limit, render the page to block import
+  if (limitIsExceeded) {
+    return (
+      <Stack gap={4}>
+        <Alert variant="danger" icon={Info}>
+          <Alert.Heading>
+            <FormattedMessage {...messages.importBlockedTitle} />
+          </Alert.Heading>
+        </Alert>
+        <FormattedMessage
+          {...messages.importBlockedBody}
+          values={{
+            limitNumber: libraryBlockLimits?.maxBlocksPerContentLibrary || 0,
+          }}
+        />
+      </Stack>
+    );
+  }
```

Comment:
It would be nice to move this inside `Banner` component.

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/2700/commits/0f0c70f466d7ca299d87e3ba3e85c82ce3aa0767